models/src/main/java/com/vimeo/networking2/Membership.kt

```diff
@@ -0,0 +1,43 @@
+package com.vimeo.networking2
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+import com.vimeo.networking2.enums.AccountType
+import com.vimeo.networking2.enums.asEnum
+
+/**
+ * This object contains user membership information.
+ */
+@JsonClass(generateAdapter = true)
+data class Membership(
+
+    /**
+     * The user's membership level
+     */
+    @Json(name = "display")
+    val display: String,
+
+    /**
+     * The user's account type.
+     */
+    @Json(name = "type")
+    val type: String,
+
+    /**
+     * Information about the user's badge.
+     */
+    @Json(name = "badge")
+    val badge: UserBadge? = null,
+
+    /**
+     * Information about the user's subscription.
+     */
+    @Json(name = "subscription")
+    val subscription: Subscription? = null
```

Comment:
Based on the spec, this field is required. Should it be non null?

Reply:
I went with nullable here because the spec is:
> `* @api-data membership.subscription (object, optional) - Information about the user's subscription.`